c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -954,7 +962,7 @@ class DeviceCachingAllocator {
     if (should_split(block, size)) {
       remaining = block;
 
-      block = new Block(device, stream, size, &pool, block->ptr);
+      block = new Block(device, stream, size, orig_size, &pool, block->ptr);
```

Comment:
Blocks can be used multiple times (in particular the not split case below). This only assigns the requested size to the block on the first use but it won't get updated in the other case. Maybe try setting the requested_bytes at the same time when the block is marked allocated since every new use of a block has to be coupled with the block being marked allocated.